### PR TITLE
Controller rspec Implemented 🧪 

### PR DESCRIPTION
**Pull request by demesameneshoa**
## In this PR I executed ☑️ ✅ 
- [x] Added `rails-controller-testing` to project Gemfile
- [x] Created Controller spec tests for `users`
- [x] Created Controller spec tests for `posts`

### CODE REVIEW

File: spec/requests/controllers_specs_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    end
+    it 'renders the posts show page for a selected post' do
+      user = create(:user) # Assuming you have FactoryBot configured
+      post = create(:post, author: user)
+      get "/users/#{user.id}/posts/#{post.id}"
+      expect(response).to have_http_status(200)
+      expect(response).to render_template(:show)
+      expect(response.body).to include('Single Post Page')
+    end
+  end
```
Comment (thecodechaser) [inline spec/requests/controllers_specs_spec.rb]: - [ ] Please make sure to place the tests in different files for users and posts controllers as requested in the project requirements.
image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @demesameneshoa👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

image

## To Highlight 🎉
- All tests are passed. ✔️
- No linter errors. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @demesameneshoa, 

To highlight 👍🏽 
✔️ Tests have been moved to individual files
✔️ All tests pass


Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by demesameneshoa** [comment]
@Ghiftee Thank you for the feedback!
🥇

**Pull request merged**